gitlab/base.py

```diff
@@ -323,6 +324,21 @@ class RequiredOptional:
     optional: Tuple[str, ...] = ()
 
 
+def custom_attrs(required: tuple = (), optional: tuple = ()) -> Callable[[F], F]:
```

Comment:
Any reason not to call it `cli_attributes` or something like that?

Reply:
My idea is to not use them only in the CLI but also in the API client itself, by inspecting these tuples to construct the request payload instead of sending them as query parameters (which will be ignored in POST/PUT as  seen in https://github.com/python-gitlab/python-gitlab/issues/1982).

So I wanted to make them more similar to `_create_attrs` and `_update_attrs` class attributes, which currently have the exact same structure. WDYT?

Reply:
So the idea is that custom methods would also reuse them not just the CLI, the way that `create()` and `update()` currently already reuse `_update_attrs` and `_create_attrs`.